### PR TITLE
Location-Watcher for Mobile & Getting Privleged Locations working on Android

### LINKED ISSUE #1811: MANAGE_EXTERNAL_STORAGE locations not addable on Mobile
### Describe the bug

Users cannot add locations locked on Android behind MANAGE_EXTERNAL_STORAGE for some reason using DocumentPicker. This issue is for a PR which helps to get this working.

For some reason, Apple allows you to add these folders easily.

### Reproduction

1. Give Spacedrive MANAGE_EXTERNAL_STORAGE permission using ADB
2. Open Spacedrive on Android
3. Click Add Location
4. Try to select Download or any other protected folder

### Expected behavior

File Explorer doesn't allow user to add folder as they are protected folders for privacy reasons.

### Platform and versions

```Shell
8.10.2
cargo 1.73.0 (9c4383fb5 2023-08-26)
rustc 1.73.0 (cc66ad468 2023-10-03)

Android 14 (One UI 6)
```

### Stack trace

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Rocky43007**
By adding a screen with a list of default locations with absolute paths as they are accessible via API calls due to the app having MANAGE_EXTERNAL_STORAGE, users can add locations which are not allowed by the file explorer on Android, like Downloads.

Closes #1811

### CODE REVIEW

**Comment by Rocky43007** [comment]
Well, Location-Watcher is basically done for iOS, but Android doesn't work yet. This will be fun.

File: apps/mobile/src/screens/settings/library/LocationSettings.tsx [hunk 1]
```diff
@@ -73,7 +73,7 @@ function LocationItem({ location, index, navigation }: LocationItemProps) {
 				<Pressable
 					style={tw`items-center justify-center rounded-md border border-app-line bg-app-button px-3 py-1.5 shadow-sm`}
 					onPress={() =>
-						fullRescan.mutate({ location_id: location.id, reidentify_objects: true })
+						fullRescan.mutate({ location_id: location.id, reidentify_objects: false }) // For some reaason, reidentify_objects being true breaks the query. But, full rescan works when it's false on mobile.
```
Comment (utkubakir) [inline apps/mobile/src/screens/settings/library/LocationSettings.tsx:76]: What kind of files do you have on your test directory? I believe when mobile's identifier runs into video files, it crashes.

Comment (Rocky43007) [reply apps/mobile/src/screens/settings/library/LocationSettings.tsx:76]: JSON files and folders.

Comment (utkubakir) [reply apps/mobile/src/screens/settings/library/LocationSettings.tsx:76]: That's interesting

**Comment by Rocky43007** [comment]
Good news, when you give the app `MANAGE_EXTERNAL_STORAGE` with `adb`, fullRescan works. But it still needs `reidentify_objects` to be false. This is with a folder with only the following files:
- .spacedrive
- a .png file
- a .pdf file

**Comment by Rocky43007** [comment]
Technically it works, and the app receives events even without the app in focus. However, library creation is borked right now.

**Comment by Rocky43007** [comment]
Hey @fogodev, do you mind taking a look at this? 

Right now, this is the flow on Android:

Library is made -> Location is added -> sd_core dies. 

From what I'm understanding, it's looking for some event or some object that's created using `LocationManager::new()`, but as `android_inotify.rs` is a function basically, it's not meeting the type requirements? 

However, the location watcher works and does receive events for directory content changes.

**Comment by utkubakir** [comment]
Posting this to our internal slack chat :)

**Comment by Rocky43007** [comment]
As a note, currently being worked on in a separate branch because I don't want to build a custom solution due to how integrated `notify-rs` is in Spacedrive.

Also, the amount of conflicting files gave me headaches, so made life easier lol.

**Comment by utkubakir** [comment]
@Rocky43007 sounds good, if you need anything you can dm me from discord too. @fogodev is our guy for the the watcher but he is indeed busy with couple of other issues these days.

**Pull request closed without merging**